### PR TITLE
Use nsymbols.

### PR DESCRIPTION
**Pull request by aartaka**
# Description

This uses the newly-abstracted `nsymbols` library to handle most of symbol resolution problems. Not much to discuss there, it's mostly deleting the code that was moved to `nsymbols`.

# Checklist:
Everything in this checklist is required for each PR.  Please do not approve a PR that does not have all of these items.

- [X] I have pulled from master before submitting this PR
- [X] There are no merge conflicts.
- [X] I've added the new dependencies as:
  - [X] ASDF dependencies,
  - [X] Git submodules,
    ```sh
	cd /path/to/nyxt/checkout
    git submodule add https://gitlab.common-lisp.net/nyxt/py-configparser _build/py-configparser
    ```
  - [x] and Guix dependencies.
- [X] My code follows the style guidelines for Common Lisp code. See:
  - [Norvig & Pitman's Tutorial on Good Lisp Programming Style (PDF)](https://www.cs.umd.edu/~nau/cmsc421/norvig-lisp-style.pdf)
  - [Google Common Lisp Style Guide](https://google.github.io/styleguide/lispguide.xml)
- [X] I have performed a self-review of my own code.
- [x] My code has been reviewed by at least one peer.  (The peer review to approve a PR counts.  The reviewer must download and test the code.)
- [X] Documentation:
  - [X] All my code has docstrings and `:documentation`s written in the aforementioned style.  (It's OK to skip the docstring for really trivial parts.)
  - [X] I have updated the existing documentation to match my changes.
  - [X] I have commented my code in hard-to-understand areas.
  - [X] I have updated the `changelog.lisp` with my changes if it's anything user-facing (new features, important bug fix, compatibility breakage).
  - [X] I have added a `migration.lisp` entry for all compatibility-breaking changes.
- [X] Compilation and tests:
  - [X] My changes generate no new warnings.
  - [X] I have added tests that prove my fix is effective or that my feature works.  (If possible.)
  - [X] New and existing unit tests pass locally with my changes.

### CODE REVIEW

File: source/mode.lisp [hunk 1]
```diff
@@ -187,51 +187,15 @@ When unset, it corresponds to the mode name."
...
-Return NIL if it's not a mode."
-  (alex:when-let ((class (find-class symbol nil)))
-    (when (mopu:subclassp class (find-class 'mode))
-      class)))
-
-(defun package-modes (packages &key (visibility :any))
-  "Return the list of mode symbols in PACKAGES.
-See `package-symbols'."
-  (delete-if (complement #'mode-class)
-             (package-symbols packages :visibility visibility)))
```
Comment (Ambrevar) [inline source/mode.lisp:202]: How can modes be found now?

Comment (aartaka) [reply source/mode.lisp:202]: With `sym:package-modes`, defined by `sym:define-symbol-type mode` :)

**Comment by Ambrevar** [comment]
CI fails with

```log
; compiling file "/home/runner/work/nyxt/nyxt/source/command.lisp" (written 31 OCT 2022 08:39:28 PM):
; 
 caught ERROR:
;   READ error during COMPILE-FILE:
;   
;     Symbol "MODE-SYMBOL-P" not found in the NSYMBOLS package.
;   
;       Line: 308, Column: -1, File-Position: 12955
;   
;       Stream: #<SB-INT:FORM-TRACKING-STREAM for "file /home/runner/work/nyxt/nyxt/source/command.lisp" {1011BAD743}>
; 
; compilation unit aborted
```

**Comment by aartaka** [comment]
Yes, that's because on version 0.1.0 of `nsymbols`, the types and predicates for those are not exported at compile-time. I'll fix some bugs, push a new release of `nsymbols`, and it should work here.

**Comment by aartaka** [comment]
I've sent a patch with nsymbols to Guix. Will get back to this PR once it's accepted.

**Comment by aartaka** [comment]
Nsymbols accepted to Guix!

**Comment by aartaka** [comment]
Ready for review. @aadcg, mind taking another look?

**Comment by aadcg** [comment]
I'd prefer to defer this one... Perhaps @Ambrevar could continue to review it since he has already start it?

**Comment by Ambrevar** [comment]
Looks good to me!

**Comment by aadcg** [comment]
2 people suffice to reach an agreement :)  Don't wait for me.

**Pull request merged**